### DISCUSSION TITLE
Release NEST Desktop as Jupyter Hub Extension

### DISCUSSION THREAD

**Discussion by jens-br** [nest-desktop/nest-desktop#480] [Ideas]
**Is your feature request related to a problem? Please describe.**
It would be helpful to have NEST Desktop as a Jupyter Hub Extension in order to use NEST Desktop on HPC resources, which are usually extremely restricted to access. However, the providers give sometimes access to a Jupyter Hub to use the resources.

**Describe the solution you'd like**
NEST Desktop as a Jupyter Hub Extension

**Describe alternatives you've considered**
Alternative ideas were iFrames or other either clumsy/manual or very complex.

**Additional context**
Example for shipping a Vue app as a Jupyter Hub Extension
 * https://github.com/studyhub-co/jupyterlab_vue , as discussed in:
 * https://discourse.jupyter.org/t/boilerplate-needed-for-making-a-jupyterlab-extension-using-vue-js/1630
 * https://github.com/jupyterlab/extension-cookiecutter-js/issues/27

Inclusion of `vue-loader` has a changed syntax:
https://github.com/vuejs/vue-loader/issues/1688#issuecomment-796603496

Maybe this is of interest as well:
https://jupyterlab.readthedocs.io/en/stable/extension/extension_dev.html#distributing-a-prebuilt-extension

**Comment by babsey** [comment]
Run it in jupyter lab
```
%%html
<iframe src="https://nest-desktop.github.io/app" width=1420 height=800></iframe>
```

**Comment by babsey** [comment]
If issued to open html file, see https://github.com/jupyterlab/jupyterlab/issues/9738

**Comment by babsey** [comment]
Copied code from https://github.com/jupyterlab/jupyterlab/issues/6235

```
from IPython.display import IFrame
IFrame(src="https://nest-desktop.github.io/app", width='100%', height='600px')
```

**Comment by babsey** [comment]
We could install `jupyterlab_iframe` (https://github.com/timkpaine/jupyterlab_iframe).
It is able to open new iframe in a jupyterlab tab displaying html content of the url https://nest-desktop.github.io/app.

Currently, a new iframe can only be opened in command palette.

By the way: There is an open PR for `jupyterlab_iframe` which enables to open iframe from launcher (https://github.com/timkpaine/jupyterlab_iframe/pull/82).

**Comment by babsey** [comment]
Close because Jens Henrik Göbbert created Jupyter Proxy Extension for NEST Desktop which runs on JSC now.

To read more: https://nest-desktop.readthedocs.io/en/latest/online-services/jupyter-jsc.html#nest-desktop-on-jupyter-jsc
